runcoms/zshenv

```diff
@@ -6,6 +6,6 @@
 #
 
 # Ensure that a non-login, non-interactive shell has a defined environment.
-if [[ "$SHLVL" -eq 1 && ! -o LOGIN && -s "${ZDOTDIR:-$HOME}/.zprofile" ]]; then
+if [[ "$SHLVL" -eq 1 && ! -o LOGIN && -s "${ZDOTDIR:-$HOME}/.zprofile" || ! "${TMPDIR}" ]]; then
```

Comment:
This check shouldn't be needed. -s checks if a file exists and has a size greater than zero, so it will be sourced if it exists.